src/modules/firewall/src/lib/Firewall.h

```diff
@@ -19,169 +24,149 @@ class FirewallLog
-    void SetDestinationPort(std::string destinationPort);
-    void SetInInterface(std::string inInterface);
-    void SetOutInterface(std::string outInterface);
-    void SetRawOptions(std::string rawOptions);
-
-    int GetRuleNum();
-    std::string GetTarget();
-    std::string GetProtocol();
-    std::string GetSource();
-    std::string GetDestination();
-    std::string GetSourcePort();
-    std::string GetDestinationPort();
-    std::string GetInInterface();
-    std::string GetOutInterface();
-    std::string GetRawOptions();
+    enum class State
+    {
+        Unknown = 0,
+        Enabled,
+        Disabled
+    };
 
-private:
-    int m_ruleNum;
-    std::string m_policy;
-    std::string m_target;
-    std::string m_protocol;
-    std::string m_source;
-    std::string m_destination;
-    std::string m_sourcePort;
-    std::string m_destinationPort;
-    std::string m_inInterface;
-    std::string m_outInterface;
-    std::string m_rawOptions;
-};
+    virtual ~GenericUtility() = default;
 
-enum FirewallStateCode
-{
-    firewallStateCodeUnknown = 0,
-    firewallStateCodeEnabled,
-    firewallStateCodeDisabled
+    virtual State Detect() const = 0;
+    virtual std::string Hash() const = 0;
 };
 
-enum UtilityStatusCode
+class IpTables : public GenericUtility
 {
-    utilityStatusCodeUnknown = 0,
-    utilityStatusCodeInstalled,
-    utilityStatusCodeNotInstalled
-};
+public:
+    typedef GenericUtility::State State;
 
-enum RuleToken
-{
-    ruleTokenNumIndex = 1,
-    ruleTokenPacketsIndex,
-    ruleTokenBytesIndex,
-    ruleTokenTargetIndex,
-    ruleTokenProtocolIndex,
-    ruleTokenOptIndex,
-    ruleTokenInIndex,
-    ruleTokenOutIndex,
-    ruleTokenSourceIndex,
-    ruleTokenDestinationIndex,
-    ruleTokenRawOptionsIndex
-};
+    IpTables() = default;
+    ~IpTables() = default;
 
-class Chain
-{
-public:
-    Chain();
-    Chain(std::string chainName);
-    ~Chain();
-    void SetChainName(std::string chainName);
-    void SetChainPolicy(std::string chainPolicy);
-    std::string GetChainName();
-    std::string GetChainPolicy();
-    int GetRuleCount();
-    std::vector<Rule*> GetRules();
-    void Append(Rule* rule);
+    State Detect() const override
+    {
+        const char* command = "iptables -S";
+
+        State state = State::Unknown;
+        char* textResult = nullptr;
+
+        if ((0 == ExecuteCommand(nullptr, command, false, false, 0, 0, &textResult, nullptr, FirewallLog::Get())))
+        {
+            if (textResult && (strlen(textResult) > 0))
+            {
+                state = State::Enabled;
+            }
+            else
+            {
+                state = State::Disabled;
+            }
+        }
+        else
+        {
+            state = State::Disabled;
+        }
+
+        FREE_MEMORY(textResult);
+
+        return state;
+    }
 
-private:
-    std::string m_chainName;
-    std::string m_chainPolicy;
-    std::vector<Rule*> m_rules;
+    std::string Hash() const override
```

Comment:
>std::string Hash() const override [](http://example.com/codeflow?start=4&length=33)

Consider skipping this wrapper and/or moving to .cpp file (too big for a header inline - having logging is enough for that)